### PR TITLE
Support CURVE25519 in `get_curve_name` when HAVE_ECC not defined

### PR DESCRIPTION
**Pull request by kaleb-himes**
Thanks to C.B. for the report.

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -16522,21 +16522,33 @@ static const char* wolfssl_ffdhe_name(word16 group)
...
+
 #ifdef HAVE_ECC
-    if (ssl->ecdhCurveOID == 0)
-        return NULL;
-    if (ssl->ecdhCurveOID == ECC_X25519_OID)
-        return "X25519";
-    return wc_ecc_get_name(wc_ecc_get_oid(ssl->ecdhCurveOID, NULL, NULL));
-#else
-    return NULL;
+        if (ssl->ecdhCurveOID != 0 && cName == NULL) {
```
Comment (dgarske) [inline src/ssl.c]: Let's not nest the logic here. Instead do a `cName == NULL` check at each point. Currently with this code if `HAVE_CURVE25519` is not defined then cName NULL check is done twice.

**Comment by kaleb-himes** [comment]
Customer Feedback was: Fix doesn't account for case when HAVE_FFDHE returns null, customer had to re-order the checks to place FFDHE lower than X25519 check to get X25519 return. Fixed with commit "Check other names when FFDHE returns NULL"

File: src/ssl.c (hunk 1 above)
Comment (kaleb-himes) [reply src/ssl.c]: Great note! Thanks.

**Comment by kaleb-himes** [comment]
Customer confirmed latest changes resolve issue.

**Pull request merged**